LanguageFeatures/Wildcards/binding_A02_t02.dart

```diff
@@ -0,0 +1,77 @@
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A local declaration whose name is `_` does not bind that name to
+/// anything. This means you can have multiple local declarations named `_` in
+/// the same namespace without a collision error.
+///
+/// @description Checks that it is not an error to have a multiple local
+/// declarations named `_`. Test parameters of constructors.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=wildcard-variables
+
+class C<T> {
+  C.constructor1(_, int _, var _) {}
+  C.constructor2(_, int _, var _, [_, int _ = 0, var _ = ""]) {}
+  C.constructor3(T _, T _, [T? _, T? _]) {}
+
+  const C.constConstructor1(_, int _, var _);
+  const C.constConstructor2(_, int _, var _, [_, int _ = 0, var _ = ""]);
+  const C.constConstructor3(T _, T _, [T? _, T? _]);
+
+  factory C.factoryConstructor1(_, int _, var _) = C.constructor1;
+  factory C.factoryConstructor2(_, int _, var _, [_, int _ = 0, var _ = ""])
+      = C.constructor2;
+  factory C.factoryConstructor3(T _, T _, [T? _, T? _]) = C.constructor3;
+}
+
+enum E<T> {
+  e1.constructor1("1", 2, "3"),
+  e2.constructor2("4", 5, "6", 7, 8, "9"),
+  e3<int>.constructor3(1, 2, null, 4])();
+
+  const E.constructor1(_, int _, var _);
+  const E.constructor2(_, int _, var _, [_, int _ = 0, var _ = ""]);
+  const E.constructor3(T _, T _, [T? _, T? _]);
+}
+
+extension type ET<T>(int id) {
+  ET.constructor1(this.id, int _, var _) {}
+  ET.constructor2(this.id, int _, var _, [_, int _ = 0, var _ = ""]) {}
+  ET.constructor3(this.id, T _, [T? _, T? _]) {}
+
+  const ET.constConstructor1(this.id, int _, var _);
+  const ET.constConstructor2(this.id, int _, var _, [_, int _ = 0, var _ = ""]);
+  const ET.constConstructor3(this.id, T _, [T? _, T? _]);
+
+  factory ET.factoryConstructor1(int _, int _, var _) = ET.constructor1;
+  factory ET.factoryConstructor2(
+      int _, int _, var _, [_, int _ = 0, var _ = ""]) = ET.constructor2;
+  factory ET.factoryConstructor3(int _, T _, [T? _, T? _]) = ET.constructor3;
+}
+
+main() {
+  C.constructor1("1", 2, true);
+  C.constructor2("1", 2, true, "3", 4, false);
+  C<int>.constructor3(1, 2, 3, 4);
+  const C.constConstructor1("1", 2, true);
+  const C.constConstructor2("1", 2, true, "3", 4, "5");
+  const C<int>.constConstructor3(1, 2, null, 4);
+  C.factoryConstructor1("1", 2, true);
+  C.factoryConstructor2("1", 2, true, "3", 4, "5");
+  C<int>.factoryConstructor3(1, 2, 3, 4);
+
+  print(E);
+
+  ET.constructor1(1, 2, "3");
+  ET.constructor2(1, 2, true, "3", 4, "5");
+  ET<int>.constructor3(1, 2, 3, 4);
+  const ET.constConstructor1("1", 2, true);
+  const ET.constConstructor2("1", 2, true, "3", 4, "5");
+  const ET<int>.constConstructor3(1, 2, null, 4);
+  ET.factoryConstructor1("1", 2, true);
+  ET.factoryConstructor2("1", 2, true, "3", 4, "5");
+  ET<String>.factoryConstructor3(1, "2", null, "4");
+}
```

Comment:
I think it would be good to cover redirecting generative constructors (`: this(...)`), too.

Another thing that would probably touch code which isn't covered by these tests is constructor tear-offs. The weird thing is that they would give rise to a situation where we are creating a function literal that calls the constructor, but we can't just reuse the parameter names of the constructor because that would yield a situation where the function literal parameters are wildcarded, and the occurrences of `_` as actual arguments in the invocation of the constructor would (probably) be undefined.

Reply:
Added tests for generative constructors.

As for the tear-offs, please clarify. What exactly is meant?
```dart
class C {
  C(_, int _);
}

main() {
  var f = C.new;
  f("a", 0); // What else can we check?
}
```

Reply:
Thanks!

About the tear-offs: The point is that they could be implemented in a way which is similar to the following:

```dart
class C {
  C(_ int _);
}

void main() {
  var f = (_, int _) => C(_, _); // Error!
}
```

and we just want to make sure the need to create a function object whose body is a constructor invocation doesn't get confused about wildcarded parameters.

So creating the tear-off and calling it should be fine!

Reply:
Thanks, it looks like all those tear-offs are in LanguageFeatures/Wildcards/binding_A01_t11.dart.